core/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java

```diff
@@ -56,7 +57,11 @@ public PerFieldMappingPostingFormatCodec(Lucene50StoredFieldsFormat.Mode compres
     public PostingsFormat getPostingsFormatForField(String field) {
         final MappedFieldType indexName = mapperService.indexName(field);
         if (indexName == null) {
-            logger.warn("no index mapper found for field: [{}] returning default postings format", field);
+            // the query terms are an percolator implementation detail and are added as Lucene indexed fields which are hidden
```

Comment:
I thought about this and my initial idea was that the percolator generates fields and the user has no control over it and therefor it should be hidden. But regardless of that this if statement is a hack... I will explore making these auto generated fields be regular mapped fields and that should allow for this if statement to be removed.